src/server.cpp

```diff
@@ -2661,6 +2653,32 @@ void Server::sendRequestedMedia(session_t peer_id,
 	}
 }
 
+void Server::stepPendingDynMediaCallbacks(float dtime)
+{
+	MutexAutoLock lock(m_env_mutex);
+
+	for (auto it = m_pending_dyn_media.begin(); it != m_pending_dyn_media.end();) {
+		it->second.expiry_timer -= dtime;
+		bool del = it->second.waiting_players.empty() || it->second.expiry_timer < 0;
+
+		if (del) {
```

Comment:
Early "return" improves readability in this case.
```C++
		if (!del) {
			// A player is still waiting. Do not cancel yet.
			it++;
			continue;
		}
```
or perhaps
`if (it->second.expiry_timer >= 0 && !it->second.waiting_players.empty()) {`